kore/src/main/java/org/kframework/SerializableSupplier.java

```diff
@@ -0,0 +1,8 @@
+// Copyright (c) Runtime Verification, Inc. All Rights Reserved.
+package org.kframework;
+
+import java.io.Serializable;
+import java.util.function.Supplier;
+
+@FunctionalInterface
+public interface SerializableSupplier<T> extends Supplier<T>, Serializable {}
```

Comment:
Where does the need for `Serializable` come in? I presume you've thought this all through, just curious what the technical reason is!

Reply:
Admittedly, this was a bit of error driven development - the original `POSet` class extended `Serializable` and some tests fail with errors about it not being serializable if I remove it! 

Making `POSet` serializable requires making the stored `Lazy<T>` serializable, which requires making the underlying `Supplier<T>` serializable.